client/server/pages/index.js

```diff
@@ -296,6 +296,8 @@ function setUpLoggedOutRoute( req, res, next ) {
 			subscriptionManagementSubkey: req.cookies.subkey,
 		};
 	}
+	const acceptLang = req.get( 'Accept-Language' )?.split( ',' )[ 0 ];
+	req.context.lang = acceptLang || req.context.lang;
```

Comment:
This isn't an expert opinion, but I also noticed there slightly more handling done for the logged-in user here:

https://github.com/Automattic/wp-calypso/blob/2dfbefa26af382afb413a2a0ac7c7c2ab898d207/client/server/pages/index.js#L376-L389

I'm not sure if that's really needed for this, though.

Also, other pages like `/themes` do locale with a path (like https://wordpress.com/es/themes). I don't know why we did that for those pages, but I'm curious if there was a specific reason that makes the /es path useful in some cases (and if that'd also be useful for this feature!)

Reply:
> Also, other pages like /themes do locale with a path (like https://wordpress.com/es/themes).

Taking a guess: maybe this has to do with how the themes page is server side rendered?

Reply:
Good point, that definitely makes it easier to cache per route. (And maybe there are related SSR benefits?) Though it'd be pretty easy to cache the language based on this too. Maybe it wasn't a thing when SSR was first added

Reply:
I noticed also that the /themes page isn't currently redirecting to the language version that's set in the browser. Yea hmm maybe it is useful for SEO too? I'm not sure